docs/source/CUSTOM-JOB.rst

```diff
@@ -0,0 +1,362 @@
+
+Currently, `deepcell.org <www.deepcell.org>`_ supports a cell tracking feature which is facilitated by the ``tracking-consumer``, which handles the multi-step process of cell tracking:
+
+  1. Send each frame of the dataset for segmentation. Frames are processed in parallel utilizing scalability and drastically reducing processing time.
+  2. Retrive model predictions and run post-processing to generate cell segmentation masks
+  3. Send cell segmentation masks for cell tracking predictions
+  4. Compile final tracking results and post for download
+
+New data processing pipelines can be implemented by writing a custom consumer. The model can be exported for tf-serving using :func:`~deepcell.utils.export_utils.export_model`.
+
+The following variables will be used throughout the setup of the custom consumer. Pick out names that are appropriate for your consumer.
+
+.. py:data:: queue_name
+
+    Specifies the queue name that will be used to identify jobs for the :mod:`redis_consumer`, e.g. ``'track'``
+
+.. py:data:: consumer_name
+
+    Name of custom consumer, e.g. ``'tracking-consumer'``
+
+.. py:data:: consumer_type
+
+    Name of consumer job, e.g. ``'tracking'``
+
+Designing a custom consumer
+---------------------------
+
+For guidance on the changes that need to be made to |kiosk-redis-consumer|, please see :doc:`kiosk-redis-consumer:CUSTOM_CONSUMER`
+
+.. |kiosk-redis-consumer| raw:: html
+
+    <tt><a href="https://github.com/vanvalenlab/kiosk-redis-consumer">kiosk-redis-consumer</a></tt>
+
+Deploying a custom consumer
+---------------------------
+
+The DeepCell Kiosk uses |helm| and |helmfile| to coordinate Docker containers. This allows the :mod:`redis_consumer` to be easily extended by setting up a new docker image with your custom consumer.
+
+1. If you do not already have an account on `Docker Hub <https://hub.docker.com/>`_. Sign in to docker in your local environment using ``docker login``.
+
+2. From the root of the ``kiosk-redis-consumer`` folder, run ``docker build <image>:<tag>`` and then ``docker push <image>:<tag>``.
+
+3. In the ``/conf/helmfile.d/`` folder in your kiosk environment, add a new helmfile following the convention ``02##.custom-consumer.yaml``. The text for the helmfile can be copied from ``0250.tracking-consumer.yaml`` as shown below. Then make the following changes to customize the helmfile to your consumer.
+
+  * Change ``releases.name`` to :data:`consumer_name`
+  * Change ``releases.values.image.repository`` and ``releases.values.image.tag``
+  * Change ``releases.values.nameOverride`` to :data:`consumer_name`
+  * Change ``releases.values.env.QUEUE`` to :data:`queue_name`
+  * Change ``releases.values.env.CONSUMER_TYPE`` to :data:`consumer_type`
+
+  .. hidden-code-block:: yaml
+    :starthidden: true
+    :label: + Show/Hide example helmfile
+
+    helmDefaults:
+      args:
+        - "--wait"
+        - "--timeout=600"
+        - "--force"
+        - "--reset-values"
+
+    releases:
+
```

Comment:
This whole section and the file are in block-quotes again.

Reply:
My personal preference would be to ignore github related formatting issues for this pull request. The primary interaction point for this documentation should be on read the docs. Figuring out how to smooth over github related formatting issues seems like it should be a marked as an issue for a future pull request.

Reply:
Fixed